src/components/sidebar/SidebarNavigation/index.tsx

```diff
@@ -25,17 +25,22 @@ const Navigation = (): ReactElement => {
 
   return (
     <SidebarList>
-      {navItems.map((item) => (
-        <ListItem key={item.href} disablePadding selected={router.pathname === item.href}>
-          <SidebarListItemButton
-            selected={router.pathname === item.href}
-            href={{ pathname: item.href, query: { safe: router.query.safe } }}
-          >
-            {item.icon && <SidebarListItemIcon badge={item.badge}>{item.icon}</SidebarListItemIcon>}
-            <SidebarListItemText bold>{item.label}</SidebarListItemText>
-          </SidebarListItemButton>
-        </ListItem>
-      ))}
+      {navItems.map((item) => {
+        const subdirectory = item.href.split('/')[1]
+        const isSelected = router.pathname.includes(subdirectory)
+
+        return (
+          <ListItem key={item.href} disablePadding selected={isSelected}>
+            <SidebarListItemButton
+              selected={isSelected}
+              href={{ pathname: item.href, query: { safe: router.query.safe } }}
+            >
+              {item.icon && <SidebarListItemIcon badge={item.badge}>{item.icon}</SidebarListItemIcon>}
+              <SidebarListItemText bold>{item.label}</SidebarListItemText>
+            </SidebarListItemButton>
+          </ListItem>
+        )
+      })}
```

Comment:
Subsections, i.e. transaction queue or module settings were not being highlighted.

Reply:
I disagree, you are in the "Transactions"/"Settings" areas of the Safe. The sidebar title would otherwise be "History" and "Setup".

`safe-react` was the same:

![image](https://user-images.githubusercontent.com/20442784/194565072-2b9dd6a7-2d0f-490d-bf1e-23e07a6cf977.png)

Reply:
🤷